compendium/test_bundles/TestBundleDSTOI22/src/Activator.cpp

```diff
@@ -0,0 +1,51 @@
+/*=============================================================================
+
+  Library: CppMicroServices
+
+  Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+  file at the top-level directory of this distribution and at
+  https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+=============================================================================*/
+#include "cppmicroservices/BundleActivator.h"
+#include "cppmicroservices/BundleContext.h"
+#include "cppmicroservices/ServiceProperties.h"
+
+#include "ServiceImpl.hpp"
+
+namespace sample
+{
+
+    cppmicroservices::BundleContext activatorProvidedCtx;
+
+    class Activator : public cppmicroservices::BundleActivator
+    {
+      public:
+        Activator() = default;
+        ~Activator() = default;
+
+        void
+        Start(cppmicroservices::BundleContext context)
+        {
+            activatorProvidedCtx = context;
```

Comment:
I'm not sure there's much value in saving the bundle context here to check later. We already guarantee that the `BundleContext` accessed through `ComponentContext` and passed to `ServiceComponent22::Activate(std::shared_ptr<ComponentContext> const& ctx)` is the same.

not storing the `BundleContext` also simplifies the test.

Reply:
Not checking this would remove a few lines, but it wouldn't really reduce complexity at all. Considering that, I think it makes sense to test all points of contact with the BundleContext at the same time.

If this is something we are already testing for, perhaps it would make sense to merge the two tests?